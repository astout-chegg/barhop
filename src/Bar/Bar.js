import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import ApiContext from '../ApiContext'
import config from '../config';

export default class Bar extends Component {
    static defaultProps = {
        handleClickDelete: () => { },
        onDeleteBar: () => { },
        match: {
            params: {}
        }
    }

    static contextType = ApiContext;

    handleClickDelete = e => {
        const barId = this.props.id
        fetch(`${config.API_ENDPOINT}/bars/` + barId, {
            method: 'DELETE',
            headers: {
                'content-type': 'application/json'
            },
            body: JSON.stringify(barId),
        })
            .then(res => {
                if(!res.ok) {
                    throw new Error(res.statusText)
                }
            })
            .then(res => {
                this.context.deleteBar(barId)
                this.props.onDeleteBar(barId)
            })
            .catch(error => {
                console.error(error)
            })
    }

    render() {
        const { name, id, address, price, rating } = this.props
        return (
            <div className='barPage'>
                <h2 className='Bar__title'>
                    <Link to={`/bar/${id}`}>
                        {name}
                    </Link>
                    <p>{address}</p>
                    <p>{price}</p>
                    <p>{rating}</p>
                    <button
                        className='Bar__delete'
                        type='button'
                        onClick={() => { this.handleClickDelete(id) }}
                    >
                        Delete
                    </button>
                </h2>
            </div>
        )    
    } 
}