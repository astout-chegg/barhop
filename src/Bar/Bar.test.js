import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Bar from './Bar'

describe(`Bar component`, () => {
    const props = {
        id: 'a',
        name: 'test-class-name',
    }

    it('renders a .Bar by default', () => {
        const wrapper = shallow(<Bar />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('renders the Bar given props', () => {
        const wrapper = shallow(<Bar {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})