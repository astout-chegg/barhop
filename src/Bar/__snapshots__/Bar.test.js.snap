// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bar component renders a .Bar by default 1`] = `
<div
  className="barPage"
>
  <h2
    className="Bar__title"
  >
    <Link
      to="/bar/undefined"
    />
  </h2>
</div>
`;

exports[`Bar component renders the Bar given props 1`] = `
<div
  className="barPage"
>
  <h2
    className="Bar__title"
  >
    <Link
      to="/bar/a"
    >
      test-class-name
    </Link>
  </h2>
</div>
`;
